src/sentry/incidents/endpoints/serializers/workflow_engine_action.py

```diff
@@ -1,19 +1,65 @@
+from collections.abc import Mapping
+from typing import Any
+
+from django.contrib.auth.models import AnonymousUser
+from django.db.models import Subquery
+
 from sentry.api.serializers import Serializer
 from sentry.incidents.endpoints.serializers.alert_rule_trigger_action import (
     get_identifier_from_action,
     get_input_channel_id,
     human_desc,
 )
-from sentry.incidents.models.alert_rule import AlertRuleTriggerAction
-from sentry.notifications.models.notificationaction import ActionService
+from sentry.notifications.models.notificationaction import ActionService, ActionTarget
 from sentry.notifications.notification_action.group_type_notification_registry.handlers.metric_alert_registry_handler import (
     MetricAlertRegistryHandler,
 )
-from sentry.workflow_engine.models import Action, ActionAlertRuleTriggerAction
+from sentry.users.models.user import User
+from sentry.users.services.user.model import RpcUser
+from sentry.workflow_engine.models import (
+    Action,
+    ActionAlertRuleTriggerAction,
+    DataCondition,
+    DataConditionAlertRuleTrigger,
+    DataConditionGroupAction,
+    DetectorWorkflow,
+    WorkflowDataConditionGroup,
+)
+from sentry.workflow_engine.models.data_condition import Condition
 
 
 class WorkflowEngineActionSerializer(Serializer):
-    def serialize(self, obj: Action, attrs, user, **kwargs):
+    def get_alert_rule_trigger_id(self, action: Action) -> int | None:
+        """
+        Fetches the alert rule trigger id for the detector trigger related to the given action
+        """
+        action_filter_data_condition = DataCondition.objects.filter(
+            condition_group__in=Subquery(
+                DataConditionGroupAction.objects.filter(action=action).values("condition_group")
+            ),
+            type=Condition.ISSUE_PRIORITY_EQUALS,
+            condition_result=True,
+        )
+        detector_workflow = DetectorWorkflow.objects.filter(
+            workflow__in=Subquery(
+                WorkflowDataConditionGroup.objects.filter(
+                    condition_group__in=Subquery(
+                        action_filter_data_condition.values("condition_group")
+                    )
+                ).values("workflow")
+            )
+        )
+        detector_trigger = DataCondition.objects.filter(
+            condition_result__in=Subquery(action_filter_data_condition.values("comparison")),
+            condition_group__in=[dw.detector.workflow_condition_group for dw in detector_workflow],
+        )
+        return DataConditionAlertRuleTrigger.objects.values_list(
+            "alert_rule_trigger_id", flat=True
+        ).get(data_condition__in=detector_trigger)
```

Comment:
why can we assume there will be a single `DataCondition` / `AlertRuleTrigger` for an `Action`? can't there be multiple `DataConditions` in a `DataConditionGroup` with `Condition.ISSUE_PRIORITY_EQUALS`?

also, i thought each `AlertRuleTrigger` can have its own set of actions, which means it should be mapped to a `DataConditionGroup`?

Reply:
This serializer will only be used by the legacy metric alert rule endpoints (I can rename it, but note it's in the `incidents/` directory). For these we can reference the [migration helpers](https://github.com/getsentry/sentry/blob/master/src/sentry/workflow_engine/migration_helpers/alert_rule.py#L583-L587) and see that there will only be one.

Reply:
i see. it might be better to connect the `DataConditionGroup` to the `AlertRuleTrigger` then, for use in this lookup